modules/coach/src/main/CoachPager.scala

```diff
@@ -18,17 +24,54 @@ final class CoachPager(
   import CoachPager._
   import BsonHandlers._
 
-  def apply(lang: Option[Lang], order: Order, page: Int): Fu[Paginator[Coach.WithUser]] = {
-    val adapter = new Adapter[Coach](
-      collection = coll,
-      selector = listableSelector ++ lang.?? { l =>
-        $doc("languages" -> l.code)
-      },
-      projection = none,
-      sort = order.predicate
-    ) mapFutureList withUsers
+  def apply(
+      lang: Option[Lang],
+      order: Order,
+      country: Option[Country],
+      page: Int
+  ): Fu[Paginator[Coach.WithUser]] = {
+    def selector = listableSelector ++ lang.?? { l => $doc("languages" -> l.code) }
+
     Paginator(
-      adapter = adapter,
+      adapter = new AdapterLike[Coach.WithUser] {
+
+        def nbResults: Fu[Int] = coll.secondaryPreferred.countSel(selector)
+
+        def slice(offset: Int, length: Int): Fu[List[Coach.WithUser]] =
+          coachRepo.coll
+            .aggregateList(length, readPreference = ReadPreference.secondaryPreferred) { framework =>
+              import framework._
+              Match(selector) -> List(
+                Sort(
+                  order match {
+                    case Alphabetical  => Ascending("_id")
+                    case NbReview      => Descending("nbReview")
+                    case LichessRating => Descending("user.rating")
+                    case Login         => Descending("user.seenAt")
+                  }
+                ),
+                PipelineOperator(
+                  $doc(
+                    "$lookup" -> $doc(
+                      "from"         -> userRepo.coll.name,
+                      "localField"   -> "_id",
+                      "foreignField" -> "_id",
+                      "as"           -> "coach"
+                    )
+                  )
+                ),
+                UnwindField("coach"),
+                Match(country.?? { c => $doc("coach.profile.country" -> c.code) })
```

Comment:
How about avoiding the whole aggregate/lookup when country is not specified? Which I guess will be the majority of queries